proposals/0279-screen-manager-subscribe-buttons.md

```diff
@@ -16,12 +16,11 @@ The proposed solution is to add new public APIs to `ScreenManager`:
 
 ### iOS
 ```objc
-typedef void (^SDLSubscribeButtonHandler)(SDLOnButtonPress *_Nullable buttonPress,  SDLOnButtonEvent *_Nullable buttonEvent, NSError *_Nullable error);
+typedef void (^SDLSubscribeButtonHandler)(SDLOnButtonPress *_Nullable buttonPress, SDLOnButtonEvent *_Nullable buttonEvent, NSError *_Nullable error);
 
-- (id<NSObject>)subscribeButton:(SDLButtonName)buttonName withBlock:(SDLRPCButtonNotificationHandler)block;
-- (void)subscribeButton:(SDLButtonName)buttonName withObserver:(id<NSObject>)observer selector:(SEL)selector;
-
-- (void)unsubscribeButtonWithObserver:(id<NSObject>)observer withCompletionHandler:(SDLScreenManagerUpdateCompletionHandler)block;
+- (nullable id<NSObject>)subscribeButton:(SDLButtonName)buttonName withUpdateHandler:(nullable SDLSubscribeButtonHandler)updateHandler;
+- (BOOL)subscribeButton:(SDLButtonName)buttonName withObserver:(id<NSObject>)observer selector:(SEL)selector;
```

Comment:
There should be method documentation here describing the selector(s) that the developer can set up, e.g.:

1. A selector with no parameters, will only be sent when a short button press occurs.
2. A selector with one parameter, (`SDLButtonName`). Will only be sent when a short button press occurs.
3. A selector with two parameters, (`SDLButtonName`, `NSError`). Will only be sent when a short button press occurs.
4. A selector with three parameters,  (`SDLButtonName`, `NSError`, `SDLOnButtonPress`). Will only be sent when any button press occurs.
5. A selector with four parameters, (`SDLButtonName`, `NSError`, `SDLOnButtonPress`, `SDLOnButtonEvent`). Will be sent if any button press occurs or any button event occurs.

What do you think about that?

If an `NSError` is passed, is the `BOOL` retval needed? The `BOOL` gives the dev no information about what went wrong. They could look at the logs, but that only works in development. The `NSError` is only useful when subscribing / unsubscribing, so if the BOOL / log is enough, then we don't need the `NSError` and can just do 0-3 parameters instead.